src/main/java/de/zalando/ep/zalenium/registry/ZaleniumRegistry.java

```diff
@@ -449,6 +450,31 @@ public void uncaughtException(Thread t, Throwable e) {
         }
     }
 
+    @Override
+    public HttpClient getHttpClient(URL url) {
+        // https://github.com/zalando/zalenium/issues/491
+        int maxTries = 3;
+        for (int i = 0; i < maxTries; i++) {
+            try {
+                HttpClient client = httpClientFactory.createClient(url);
+                if (i > 0) {
+                    String message = String.format("Successfully created HttpClient for url %s, after attempt #%s", url, (i+1));
+                    LOG.warn(message);
+                }
+                return client;
+            } catch (Exception | AssertionError e) {
+                String message = String.format("Error while getting the HttpClient for url %s, attempt #%s", url, (i+1));
+                LOG.debug(message, e);
+                if (i == 2) {
+                    throw e;
+                }
```

Comment:
So, you don't think there should be a random sleep? Just retrying it straight away seems to fix it in all cases?

Reply:
It worked right away in all cases I tried. But I agree with you, it is a good idea to add a small delay, random between 1 to 5 seconds.